pkg/master/reconcilers/lease.go

```diff
@@ -119,16 +119,16 @@ func NewLeases(storage storage.Interface, baseKey string, leaseTime time.Duratio
 }
 
 type leaseEndpointReconciler struct {
-	endpointStorage       rest.StandardStorage
+	endpointClient        corev1client.EndpointsGetter
```

Comment:
in `MasterCount` reconciler we are using a loopback client to manage endpoints while in `Lease` its using direct etcd connection. is that inconsistency designed? i suppose we dont need to value speed here when reconciling endpoints so this pull is replacing rest storage w/ clientset. wdyt, or the other way round? @deads2k

Reply:
> i suppose we dont need to value speed here

do we? not sure about it. valuing speed looks also reasonable here

Reply:
> do we? not sure about it. valuing speed looks also reasonable here

This is actually fairly slow moving.  When you consider the improvement in reasoning about a client versus a secondary, parallel storage mechanism, this is a clear win.